### DISCUSSION TITLE
Can PSRules be run against terraform templates for Azure resources

### DISCUSSION THREAD

**Discussion by vishalgoyal16444** [microsoft/PSRule#995] [Q&A] [answered]
We have several repos created and hosted on our Azure DevOps. These repos have terraform scripts to deploy azure resources. Can we run PSRules against these repositories.

Any links to documentation on how to do this would be very helpful.

**Comment by BernieWhite** [comment, marked as answer]
@vishalgoyal16444 In short, you can run PSRule against deployed Terraform resources but not against HCL infrastructure as code. It has been requested, but we are still in the process of investigating if we can support this because of the way Terraform modules are built.

Have a look at this https://azure.github.io/PSRule.Rules.Azure/faq/#isterraformsupported for more details.

**Reply by vishalgoyal16444** [reply]
Thanks @BernieWhite. Support for HCL terraform IaaC scripts will be very useful as this allows to validate before deploying resources.
